plugins/woocommerce-blocks/assets/js/blocks/product-template/edit.tsx

```diff
@@ -35,6 +38,16 @@ import './editor.scss';
 import { getDefaultStockStatuses } from '../product-collection/constants';
 
 const DEFAULT_QUERY_CONTEXT_ATTRIBUTES = [ 'collection' ];
+const getNoResultsText = ( collection?: CoreCollectionNames | string ) => {
+	if ( collection === CoreCollectionNames.HAND_PICKED ) {
+		return __(
+			'No products chosen yet. Use "Hand-Picked Products" in Inspector Controls to choose products to display',
```

Comment:
Should the product template be tangled up with a specific collection like that? Maybe we could pass this custom 'no results' text via the Hand-Picked collection attributes or sth? @kmanijak @imanish003

Reply:
After further inspection, I think we should drop this custom 'no results' message entirely. The reason is that it will also show this message when we hand-pick products but then apply filters that clear the results (no intersection).

As an alternative, I suggest improving the wording of the current message so that it's more informative and matches the wider context so that we can use it for all cases:

**Current message**:

`No results found.`

**Suggested new message**:

_Choose one, but we can mix and match. 😉_

1. `No products to display. Check your filter settings in the block settings panel.`
2. `No products found. Adjust filters in the block settings to populate this collection.`
3. `No products are currently displayed. Try adjusting the filters in the block settings panel.`

> [!IMPORTANT]
> I used the `settings panel` name from the [block editor handbook](https://developer.wordpress.org/block-editor/) nomenclature. I think we should avoid using `inspector controls` as it's not a user-facing name. 

cc: @j111q

Reply:
> The reason is that it will also show this message when we hand-pick products but then apply filters that clear the results (no intersection).

Aha! Good catch.

> As an alternative, I suggest improving the wording of the current message so that it's more informative and matches the wider context so that we can use it for all cases:

I really like your suggestion to improve the general 'no results' message. It provides more context and guidance to the user. Here is my suggestion: "No products found. Try adjusting the filters in the block settings panel if there are products in your store."

- It suggests adjusting filters if there are indeed products in the store that aren't showing up due to the current filter settings.
- It implicitly acknowledges that if there are no products in the store at all, adjusting filters won't help.

What do you think?

@WunderBart As we need to merge this PR on Monday, probably we can go with any of the above. If needed, we can iterate on the message in follow-up PR. what do you think?

Reply:
Thanks for the input, Manish! Two thoughts on the suggested wording:

- For the first part, I think we should go with `No products to display`. It matches both situations where filters are and are not applied, while `No products found` implies that a search has already been performed (misleading in the case of the initial Hand-Picked collection state).
- For the second part, I'd skip the `if there are products in your store` bit because it creates a precedent to be followed in all other product-related blocks. I think it should be implied that products must be available to use product-related blocks.

To sum up, I'd go with:

`No products to display. Try adjusting the filters in the block settings panel.`

WDYT?

Reply:
> As we need to merge this PR on Monday, probably we can go with any of the above. If needed, we can iterate on the message in follow-up PR. what do you think?

As per the above, I went ahead and updated the copy in 6e2ddf447e53c4eaab3770c6bd603b4100dd3c31.

Reply:
Makes sense to me. Thanks for updating the copy 🙌🏻